addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/handler/DSCAlarmBaseBridgeHandler.java

```diff
@@ -540,21 +541,87 @@ public synchronized void handleIncomingMessage(String incomingMessage) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        logger.warn("No bridge commands defined.");
+
+        logger.debug("handleCommand(): Command Received - {} {}.", channelUID, command);
+
+        if (command instanceof RefreshType) {
+            return;
+        }
+
         if (isConnected()) {
             switch (channelUID.getId()) {
                 case BRIDGE_RESET:
                     if (command == OnOffType.OFF) {
                         disconnect();
                     }
                     break;
+                case SEND_COMMAND:
+                    if (!command.toString().isEmpty()) {
+                        String[] tokens = command.toString().split(",");
+
+                        String cmd = tokens[0];
+                        String data = "";
+                        if (tokens.length > 1) {
+                            data = tokens[1];
+                        }
+
+                        sendDSCAlarmCommand(cmd, data);
+
+                        updateState(channelUID, new StringType(""));
+                    }
+                    break;
                 default:
                     break;
             }
         }
     }
 
     /**
+     * Method to send a sequence of key presses one at a time using the '070' command.
+     *
+     * @param keySequence
+     */
+    private boolean sendKeySequence(String keySequence) {
+        logger.debug("sendKeySequence(): Sending key sequence '{}'.", keySequence);
+
+        boolean sent = false;
+
+        for (char key : keySequence.toCharArray()) {
+            sent = sendCommand(DSCAlarmCode.KeyStroke, String.valueOf(key));
+
+            if (!sent) {
+                return sent;
+            }
+        }
+
+        return sent;
+    }
+
+    /**
+     * Sends a DSC Alarm command
+     *
+     * @param command
+     * @param data
+     */
+    public boolean sendDSCAlarmCommand(String command, String data) {
+        logger.debug("sendDSCAlarmCommand(): Attempting to send DSC Alarm Command: command - {} - data: {}", command,
+                data);
+
+        try {
+            DSCAlarmCode dscAlarmCode = DSCAlarmCode.getDSCAlarmCodeValue(command);
+
+            if (dscAlarmProtocol.equals(DSCAlarmProtocol.IT100_API) && dscAlarmCode.equals(DSCAlarmCode.KeySequence)) {
+                return sendKeySequence(data);
+            } else {
+                return sendCommand(dscAlarmCode, data);
+            }
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception